src/NeXusDataset/ExtensibleDataset.h

```diff
@@ -14,7 +14,7 @@
 
 #pragma once
 
-#include "logger.h"
+#include "../logger.h"
```

Comment:
I thought that the _src_ directory was set as an include directory but if it is not, it should be done so as to avoid relative include paths like this one.

Reply:
I made a conscious decision not to do that because I thought it would encourage people to consider whether they should be adding dependency with other parts of the codebase. But I could be persuaded otherwise...

Reply:
I can see your reasoning (though I do not believe it will make much of a difference). My reason for disliking relative paths when including headers is because I feel that it makes it harder to (manually)  find files in a repository and it makes re-factoring of code a bit more tedious. But to be honest it is not that bad if you only have to go back one level (i.e. "../"). It is worse when there are several levels of relative paths (e.g. "../../" or "../some_path/").